src/allotropy/parsers/qiacuity_dpcr/qiacuity_dpcr_parser.py

```diff
@@ -0,0 +1,248 @@
+            well_item, SAMPLE_TYPE_COLUMN_NAME, required=False
+        )
+        # TODO: When the sample role type model is updated in this repo, we should update this
+        # Map sample role types to valid sample role types from ASM
+        if sample_role_type is not None:
+            sample_role_type = SAMPLE_ROLE_TYPE_MAPPING[sample_role_type]
+            sample_document.sample_role_type = sample_role_type
+
+        well_location_identifier = self._get_value_by_col_name(
+            well_item, WELL_COLUMN_NAME, required=False
+        )
+        # TODO: Make a helper for all of these "only set if not none" optional document fields?
+        if well_location_identifier is not None:
+            sample_document.well_location_identifier = well_location_identifier
+
+        # ASM expects a string- ok to cast?
+        well_plate_identifier = str(
+            self._get_value_by_col_name(
+                well_item, WELL_PLATE_IDENTIFIER_COLUMN_NAME, required=False
+            )
+        )
+        if well_plate_identifier is not None:
+            sample_document.well_plate_identifier = well_plate_identifier
+        return sample_document
+
+    def _get_device_control_document(self) -> DeviceControlDocumentItem:
+        return DeviceControlDocumentItem(
+            device_type=DEVICE_TYPE, device_identifier=DEVICE_IDENTIFIER
+        )
+
+    def _get_processed_data_document(
+        self, well_item: pd.Series
+    ) -> ProcessedDataDocumentItem:
+        number_concentration = TQuantityValueNumberPerMicroliter(
+            value=self._get_value_by_col_name(
+                well_item, CONCENTRATION_COLUMN_NAME, required=True
+            )
+        )
+        positive_partition_count = TQuantityValueNumber(
+            value=self._get_value_by_col_name(
+                well_item, POSITIVE_COUNT_COLUMN_NAME, required=True
+            )
+        )
+        processed_data_document = ProcessedDataDocumentItem(
+            number_concentration=number_concentration,
+            positive_partition_count=positive_partition_count,
+        )
+        # Negative partition count is optional
+        negative_partition_count = TQuantityValueNumber(
+            value=self._get_value_by_col_name(
+                well_item, NEGATIVE_COUNT_COLUMN_NAME, required=False
+            )
+        )
+        if negative_partition_count is not None:
+            processed_data_document.negative_partition_count = negative_partition_count
+        return processed_data_document
+
+    def _get_data_processing_document(
+        self, well_item: pd.Series
+    ) -> DataProcessingDocument:
+        return DataProcessingDocument(
+            flourescence_intensity_threshold_setting=TQuantityValueUnitless(
+                value=self._get_value_by_col_name(
+                    well_item, FIT_SETTING_COLUMN_NAME, required=False
+                )
+            )
+        )
+
+    def _get_value_by_col_name(
```

Comment:
Should should be able to replace this with `try_str_from_series` (when required=True) and `try_str_from_series_or_none` (when required=False)

Reply:
Thank you! Looks like this was the cause of my lint failure

Reply:
This is updated, resolving the conversation